include/votca/tools/graphnode.h

```diff
@@ -63,6 +63,10 @@ class GraphNode {
   void setDouble(const std::unordered_map<std::string, double> double_vals);
   void setStr(const std::unordered_map<std::string, std::string> str_vals);
 
+  void addInt(std::string label, const Index& value);
+  void addDouble(std::string label, const double& value);
+  void addStr(std::string label, const std::string& value);
+
```

Comment:
concerning how to carry values. I think if you want to make it general, template the whole thing on the value type. Then people can create their own structs etc...